lib/controller.js

```diff
@@ -451,6 +451,45 @@ controller.printAvailableNetworks = function(opts) {
     });
 };
 
+controller.getWifiInfo = function(opts) {
+  return Tessel.get(opts)
+    .then(function(selectedTessel) {
+      return selectedTessel.getWifiInfo()
+        .then(function(network) {
+          // Grab inet lines, flatmap them, remove empty
+          // Wanted to do this with awk and cut inside commands.js
+          var ips = network.ips.filter(function(item) {
+              return /inet/.exec(item);
+            })
+            .map(function(line) {
+              return line.split(' ');
+            })
+            .reduce(function(a, b) {
+              return a.concat(b);
+            })
+            .filter(function(item) {
+              return /addr/.exec(item);
+            })
+            .map(function(chunk) {
+              return chunk.split(':')[1];
+            })
+            .filter(function(addr) {
+              return addr.length;
+            });
+
```

Comment:
If the intent of the code is clear, why consolidate

I can replace the last three with the following
```js
.reduce(function(list, ip) {
  if (/addr/.exec(item)) {
    var ip = chunk.split(':')[1];
    if (ip.length) list.push[ip];
  }
  return list;
});
```

But it doesn't make it more clear (to me) what is happening.